internal/component/loki/secretfilter/secretfilter.go

```diff
@@ -254,9 +254,14 @@ func (c *Component) redactLine(line string, secret string, ruleName string) stri
 }
 
 func hashSecret(secret string) string {
-	hasher := sha1.New()
+	hasher := sha256.New()
 	hasher.Write([]byte(secret))
-	return fmt.Sprintf("%x", hasher.Sum(nil))
+	hashBytes := hasher.Sum(nil)
+
+	// Only use the first half of the hash
+	firstHalf := hashBytes[:len(hashBytes)/2]
```

Comment:
This sort of feels odd, could we use a native 128bit hash? This feels harder to use.